src/WheelsAndGoods.DataAccess/Repositories/OrdersRepository.cs

```diff
@@ -13,14 +14,43 @@ public OrdersRepository(DatabaseContext databaseContext)
         {
         }
 
-        public async Task<IReadOnlyCollection<Order>> GetOrders()
+        public async Task<IReadOnlyCollection<Order>> GetOrders(FilterOrderModel filterOrderModel)
         {
-            return await Context.Orders
+            var query = Context.Orders
                 .Include(order => order.Customer)
+                .AsQueryable();
+
+            if (filterOrderModel.From != null)
+            {
+                query = query.Where(order => EF.Functions.ILike(
+                    order.From.ToLower(), $"%{filterOrderModel.From.ToLower()}%"));
+            }
+
+            if (filterOrderModel.Title != null)
+            {
+                query = query.Where(order => EF.Functions.ILike(
+                    order.Title.ToLower(), $"%{filterOrderModel.Title.ToLower()}%"));
+            }
+
+            if (filterOrderModel.To != null)
+            {
+                query = query.Where(order => EF.Functions.ILike(
+                    order.To.ToLower(), $"%{filterOrderModel.To.ToLower()}%"));
+            }
+
+            if (filterOrderModel.CustomerFullName != null)
+            {
+                query = query.Where(order => EF.Functions.ILike(
+                    $"{order.Customer.Firstname} {order.Customer.Lastname}".ToLower(),
+                    $"%{filterOrderModel.CustomerFullName.ToLower()}%"));
+            }
+
+            return await query.Where(order => order.Price == filterOrderModel.Price)
```

Comment:
According to requirements, Price is also optional. Moreover, the filter is not '=='.